ta/pkcs11/src/object.c

```diff
@@ -538,8 +563,19 @@ enum pkcs11_rc entry_find_objects_init(struct pkcs11_client *client,
 	 * candidates that match caller attributes.
 	 */
 
-	LIST_FOREACH(obj, &session->object_list, link) {
-		if (check_access_attrs_against_token(session, obj->attributes))
+	soh_db = get_object_handle_db(session);
+
+	/* scan all session objects first */
+	for (i = 0; i < soh_db->max_ptrs; i++) {
+		obj = soh_db->ptrs[i];
+		if (!obj)
+			continue;
```

Comment:
Instead of session object db, can we browse over session list instead

TAILQ_FOREACH(sess, get_session_list(session), link) {
                if (sess->token != session->token)
                        continue;

                LIST_FOREACH(obj, &sess->object_list, link) {
<code below goes here >
                }

Reply:
I'll add a review commit with this.